src/ports/postgres/modules/mxgboost/madlib_xgboost.py_in

```diff
@@ -0,0 +1,549 @@
+
+    grid_size = len(params_grid)
+    sql = """
+        CREATE TEMP TABLE {grid_search_params_temp_tbl_df}
+        AS
+        (
+            SELECT
+                df,
+                generate_series(1, {grid_size}) AS params_index
+            FROM
+            (
+                SELECT
+                    {schema_madlib}.__serialize_pandas_dframe_as_bytea__(
+                        '{source_table}',
+                        '{id_column}',
+                        '{class_label}',
+                        '{list_of_features}'
+                    ) AS df
+            )q
+        ) {dist_str}
+    """.format(**locals())
+    plpy.execute(sql)
+
+    #Invoke XGBoost's train by passing each row from parameter list table. This will run in parallel.
+    grid_search_results_tbl_tmp = unique_string('results_tbl')
+    features_str = features = str(features).replace('[','').replace(']','').replace(',',',\n')
+    sql = """
+        CREATE TEMP TABLE {grid_search_results_tbl_tmp}
+        AS
+        (
+            SELECT
+                {schema_madlib}.__xgboost_train_parallel__(
+                    df,
+                    ARRAY[
+                        {features_str}
+                    ],
+                    '{class_label}',
+                    params,
+                    $CW${class_weights}$CW$,
+                    {train_set_size},
+                    '{id_column}',
+                    '{train_set_split_var}'
+                ) AS mdl_results,
+                t1.params_index
+            FROM
+                {grid_search_params_temp_tbl} t1,
+                {grid_search_params_temp_tbl_df} t2
+            WHERE
+                t1.params_index = t2.params_index
+        ) {dist_str}
+    """.format(**locals())
+    plpy.execute(sql)
+
+    sql = """
+        CREATE TABLE {grid_search_results_tbl_summary}
+        AS
+        (
+            SELECT
+                now() AS mdl_train_ts,
+                '{source_table}'||'_xgboost' AS mdl_name,
+                (mdl_results).features,
+                (mdl_results).params,
+                (mdl_results).fnames,
+                (mdl_results).importance,
+                (mdl_results).precision,
+                (mdl_results).recall,
+                (mdl_results).fscore,
+                (mdl_results).support,
+                (mdl_results).test_ids::INTEGER[],
+                params_index
+            FROM
+                {grid_search_results_tbl_tmp}
+
+        ) {dist_str}
+    """.format(**locals())
+    plpy.execute(sql)
+
+    sql = """
+        CREATE TABLE {grid_search_results_tbl}
+        AS
+        (
+            SELECT
+                (mdl_results).mdl AS model,
+                (mdl_results).features,
+                params_index
+            FROM
+                {grid_search_results_tbl_tmp}
+
+        ) {dist_str}
+    """.format(**locals())
+    plpy.execute(sql)
+
+def xgboost_predict(schema_madlib, scoring_tbl, mdl_table, mdl_output_tbl,
```

Comment:
do you think it would be useful to move this in a separate file, just for maintainability. or maybe we can do that later as this file grows